src/easylink/graph_components.py

```diff
@@ -41,20 +42,48 @@ class InputSlot:
     env_var: str | None
     """The environment variable that is used to pass a list  of data filepaths to 
     an ``Implementation``."""
-    validator: Callable[[str], None]
+    validator: Callable[[str], None] = field(compare=False)
     """A function that validates the input data being passed into the pipeline via 
     this ``InputSlot``. If the data is invalid, the function should raise an exception 
     with a descriptive error message which will then be reported to the user.
     **Note that the function *must* be defined in the** :mod:`easylink.utilities.validation_utils` 
     **module!**"""
-    splitter: Callable[[list[str], str, Any], None] | None = None
+    splitter: Callable[[list[str], str, Any], None] | None = field(
```

Comment:
This isn't strictly necessary since we're explicitly only using the __name__ of callables, but it seems useful to communicate to future readers

Reply:
What does `field` do with these arguments?

Reply:
a dataclass `field` just allows for additional metadata. In this case, I'm using it to relay to readers that these arguments are not being used in the hashing (well, at least wholly used).